hnn_core/tests/test_network.py

```diff
@@ -933,6 +935,32 @@ def test_synaptic_gains():
         else:
             assert conn['nc_dict']['gain'] == 1.0
 
+    # Two argument check
+    net.update_weights(i_e=0.5, i_i=0.25, copy=False)
+    for conn in net.connectivity:
+        if (conn['src_type'] in i_cell_names and
+                conn['target_type'] in e_cell_names):
+            assert conn['nc_dict']['gain'] == 0.5
+        elif (conn['src_type'] in i_cell_names and
+                conn['target_type'] in i_cell_names):
+            assert conn['nc_dict']['gain'] == 0.25
+        else:
+            assert conn['nc_dict']['gain'] == 1.0
+
+    nb_updated = NetworkBuilder(net)
+    # Check weights are altered
+    def _get_weight(nb, conn_name, idx=0):
+        return nb.ncs[conn_name][idx].weight[0]
+    # i_e check
+    assert (_get_weight(nb_updated, 'L2Basket_L2Pyr_gabaa') /
+            _get_weight(nb_base, 'L2Basket_L2Pyr_gabaa')) == 0.5
+    # i_i check
+    assert (_get_weight(nb_updated, 'L2Basket_L2Basket_gabaa') /
+            _get_weight(nb_base, 'L2Basket_L2Basket_gabaa')) == 0.25
+    # Unaltered check
+    assert (_get_weight(nb_updated, 'L2Pyr_L5Basket_ampa') /
+            _get_weight(nb_base, 'L2Pyr_L5Basket_ampa')) == 1
```

Comment:
I found that the NetworkBuilder has weights that are not adjusted by the distance calculation. It seems like they are just the weights * gain and so the quotient of the updated weight divided by the original weight is the gain constant. 

This is just a spot check of 3 connections. We could write a more comprehensive test if you think it would be better.

Reply:
The `idx` argument of the `_get_weight` function allows you to index other cells in the list. I tried this with other cells and their weights appeared adjusted but the ratio still adheres to the gain adjustment, which may suggest that the gain is being applied linearly?  

```
>print(_get_weight(nb_updated, 'L2Basket_L2Pyr_gabaa', 1))
 print(_get_weight(nb_base, 'L2Basket_L2Pyr_gabaa', 1))
Out[4]:
0.024990001999733363
0.04998000399946673
>0.024990001999733363 / 0.04998000399946673
Out[5]: 0.5
```

Reply:
@gtdang I was wrong about how the gaussian is used! Here's the equation:
https://github.com/jonescompneurolab/hnn-core/blob/818b714b3452206362079d941e43953bf2bca480/hnn_core/cell.py#L53

`height` is actually where the `weight` variable is passed, so multiplying by a gain factor is indeed linear. The gaussian is just used as a multiplying factor where the number gets smaller with increasing cell-cell distances.